src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -1007,6 +1053,49 @@ type internal TypeCheckInfo
                     Some (List.map ItemWithNoInst items, denv, m)
                     |> Option.map toCompletionItems
 
+            // Completion at ' match x with S... -> () ' unless we're dotting into a type/module/...
+            | Some (CompletionContext.Match (MatchContext.ClausePatternIdentifier range)) when origLongIdentOpt.IsNone || origLongIdentOpt.Value.IsEmpty ->
+                match GetUnionCasesAndActivePatternsEnvironmentLookupResolutionsAtPosition range.End with
```

Comment:
Please factor these cases of `GetDeclItemsForNamesAtPosition` out into separate functions